libs/grok/src/main/java/org/elasticsearch/grok/Grok.java

```diff
@@ -220,8 +241,19 @@ public boolean match(String text) {
         byte[] textAsBytes = text.getBytes(StandardCharsets.UTF_8);
         Map<String, Object> fields = new HashMap<>();
         Matcher matcher = compiledExpression.matcher(textAsBytes);
-        int result = matcher.search(0, textAsBytes.length, Option.DEFAULT);
-        if (result != -1 && compiledExpression.numberOfNames() > 0) {
+        int result;
+        try {
+            threadInterrupter.register();
+            result = matcher.search(0, textAsBytes.length, Option.DEFAULT);
+        } finally {
+            threadInterrupter.deregister();
+        }
+        if (result == Matcher.INTERRUPTED) {
+            throw new IllegalArgumentException("grok pattern matching is too complex and takes too long to execute");
+        } else if (result == Matcher.FAILED) {
+            // I think we should throw an error here?
```

Comment:
and change current behavior?

Reply:
Agreed, this should be done in a follow up.

I don't think is different behaviour, because in `GrokProcessor` we throw an error when `null` is returned.